core/trino-main/src/test/java/io/trino/operator/output/TestPositionsAppender.java

```diff
@@ -215,6 +216,23 @@ public void testConsecutiveBuilds(Type type)
         assertEquals(positionsAppender.build().getPositionCount(), 0);
     }
 
+    // testcase for jit bug described https://github.com/trinodb/trino/issues/12821.
+    // this test needs to be run first (hence lowest priority) as order of tests
+    // influence jit compilation making this problem to not occur if other tests are run first.
+    @Test(priority = Integer.MIN_VALUE)
```

Comment:
Global test ordering isn't effective in preventing JIT from doing something.
For example, someone else may need a similar test and will also add `@Test(priority...)` removing test coverage.

I am not convinced we need this test (since it's a JVM problem).
If we do, however, maybe this needs to be in a separate profile, so it runs in isolation.

Reply:
There is a tradeoff here. Since it's unlikely that this will happen again (although it can happen when migrating to new java versions), there is little sense in investing much time and complexity into this test. I don't know how much complexity a separate profile adds so this may be a non-issue.
But also, in practice it works and even if in the future it won't work during CI, one can always run it in isolation to confirm this issue if it's suspected.